adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -189,4 +160,46 @@ - (void)unityAdsAdLoaded:(nonnull NSString *)placementId {
   }
 }
 
+#pragma mark - UnityAdsShowDelegate Methods
+
+- (void)unityAdsShowClick:(NSString *)placementId {
+  // The Unity Ads SDK doesn't provide an event for leaving the application, so the adapter assumes
+  // that a click event indicates the user is leaving the application for a browser or deeplink, and
+  // notifies the Google Mobile Ads SDK accordingly.
+  [_adEventDelegate reportClick];
+}
+
+- (void)unityAdsShowComplete:(NSString *)placementId withFinishState:(UnityAdsShowCompletionState)state {
+  if (state == kUnityAdsFinishStateCompleted) {
+    [_adEventDelegate didEndVideo];
+
+    // Unity Ads doesn't provide a way to set the reward on their front-end. Default to a reward
+    // amount of 1. Publishers using this adapter should override the reward on the AdMob
+    // front-end.
+    GADAdReward *reward = [[GADAdReward alloc] initWithRewardType:@""
+                                                     rewardAmount:[NSDecimalNumber one]];
+    [_adEventDelegate didRewardUserWithReward:reward];
+  } else if (state == kUnityAdsFinishStateError) {
+    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
+        GADMAdapterUnityErrorFinish,
+        @"UnityAds finished presenting with error state kUnityAdsFinishStateError.");
+    [_adEventDelegate didFailToPresentWithError:error];
+  }
+
+  [_adEventDelegate willDismissFullScreenView];
+  [_adEventDelegate didDismissFullScreenView];
+}
+
+- (void)unityAdsShowFailed:(NSString *)placementId withError:(UnityAdsShowError)error withMessage:(NSString *)message {
+  if (_adEventDelegate) {
+    NSError *errorWithDescription =
```

Comment:
I know its not your code. You just moved it probably
I find this pattern a bit weird, Id rather use constants since we dont actually use those error enums as they are.

Reply:
Yeah, I most of this code is just moved from the old `UnityAdsExtendedDelegate` methods to the new `UnityAdsShowDelegate` methods.

I'll take a look at it and see if it can be cleaned up some.